source/codegen/templates/proto_helpers.mako

```diff
@@ -30,16 +30,22 @@ enum ${service_class_prefix}Attributes {
 <%
   enum = enums[enum_name]
   enum_value_prefix = enum.get("enum-value-prefix", common_helpers.pascal_to_snake(enum_name).upper())
+  is_mapped_enum = enum.get("generate-mappings", False)
+  enum_name = f"{enum_name}Mapped" if (is_mapped_enum and not enum_name.endswith("Mapped")) else enum_name
   allow_alias = enum.get("allow-alias", proto_helpers.should_allow_alias(enum))
   nonint_index = 1
 %>\
 enum ${enum_name} {
 %   if allow_alias == True:
   option allow_alias = true;
 %   endif
+%   if is_mapped_enum:
+  ${enum_value_prefix}_MAPPED_UNSPECIFIED = 0;
```

Comment:
Is there a reason we added Mapped suffix explicitly for UNSPECIFIED enums?

Reply:
Yes, `MAPPED` suffix is added intentionally. If you take case of `NiDmmInt32AttributeValues` and `NiDmmInt32AttributeValuesMapped` enums, both of them will have `NIDMM_INT32_UNSPECIFIED` constant if we don't add `MAPPED` substring. Multiple enum constants with same name is not allowed in proto.